upgrade_test.py

```diff
@@ -1146,3 +1148,29 @@ def _update_argus_upgraded_version(self, node, new_version):
                 self.log.warning("Couldn't extract version from %s", new_version)
         except Exception as exc:  # pylint: disable=broad-except
             self.log.exception("Failed to save upgraded Scylla version in Argus", exc_info=exc)
+
+    def _update_os_packages_on_preinstalled_scylla_images(self, node: BaseNode, package: str) -> None:
+        """Update packages according to procedure in docs.
+
+        https://docs.scylladb.com/stable/upgrade/upgrade-opensource/upgrade-guide-from-4.5-to-4.6/
+        upgrade-guide-from-4.5-to-4.6-ubuntu-20-04#update-3rd-party-and-os-packages"""
+        if not self.params.get("use_preinstalled_scylla"):
+            self.log.info("Skipping OS packages upgrade as this is not preinstalled Scylla image")
+            return
+        if not node.distro.is_ubuntu20:
+            self.log.info("Skipping OS packages upgrade as it's supported only on Ubuntu20")
+            return
+        self.log.info("Upgrading OS packages")
+        package_path = self._download_packages_file(node, package)
+        node.remoter.run(f"cat {package_path} | sudo xargs -n1 apt-get -y")
+
+    def _download_packages_file(self, node: BaseNode, package: str) -> str:
+        """Downloads scylla-packages file used for scylla and OS packages upgrade in AMI's.
+        Returns path to packages file."""
+        pkg, ver, url, _, arch, _ = node.remoter.run(f"apt-cache madison {package}").stdout.replace("|", "").split()
+        arch = "x86_64" if arch == "amd64" else "aarch64"
```

Comment:
IIUC the logic here is inverted:

```
In [1]: arch = 'amd64'                                                                                                                                                                                                                                                            

In [2]: arch = "x86_64" if arch == "amd64" else "aarch64"                                                                                                                                                                                                                         

In [3]: print(arch)                                                                                                                                                                                                                                                                
x86_64
```

Reply:
`x86_64` is `amd64`, that's sound correct to me.. (i.e. different tools call it different names)

Reply:
Anyhow, one should be mapping those

```python
mapping = dict(amd64='x86_64, arm64='aarch64')
```

Reply:
yes, you are right, i misread it

Reply:
when you run `apt-cache madison {package}` it returns something like this:
```
scylla | 5.2.0~dev-0.20220830.4e413787d2dd-1 | http://downloads.scylladb.com/unstable/scylla/master/deb/unified/2022-08-30T04:54:00Z/scylladb-master stable/main amd64 Packages
```

Reply:
Why aren't we following the instructions? And installing straight out of file ?

Also why are we downloading on our own, and not letting apt do it ? (We skip the retrying and different mirror as fallback, and any other logic apt has)

Reply:
This is only how I get `scylla-packages` file - which will be provided as link in docs (I saw some PR with it, but not merged yet) and apt is not downloading it. This way it should work also on master branch. 

Packages installation is done exactly like in docs.